theano/gof/op.py

```diff
@@ -694,6 +694,9 @@ def __ne__(self, other):
     # Python implementation #
     #########################
 
+    def grad2(self, inputs, outputs, output_grads):
```

Comment:
Also, it would be possible to do a number of yucky things to accept both signatures, (the simplest being a try..except in the theano.gradient method, like what was done to add the c_compiler argument), but I don't like complexifiying the call protocol like that.  It's better to have a single name with a single signature.

In the end if we really want to keep the grad() name, we could add a method called _grad() that would try both signatures on public method and returns what works.